src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -70,6 +70,129 @@ public static NuGetFramework Parse(string folderName, IFrameworkNameProvider map
+                    }
+                }
+
+                if (version.Major >= 5
+                    && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.Net, framework))
+                {
+                    framework = FrameworkConstants.FrameworkIdentifiers.NetCoreApp;
+                }
+
+                if (StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.Portable, framework)
+                    && !string.IsNullOrEmpty(profile)
+                    && profile.Contains("-"))
+                {
+                    // Frameworks within the portable profile are not allowed
+                    // to have profiles themselves #1869
+                    throw new ArgumentException(string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.InvalidPortableFrameworksDueToHyphen,
+                        profile));
+                }
+
+                if (!mappings.TryGetProfile(framework, targetFrameworkProfile ?? string.Empty, out profile))
+                {
+                    profile = targetFrameworkProfile ?? string.Empty;
+                }
+
+                if (StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.Portable, framework))
+                {
+                    IEnumerable<NuGetFramework> clientFrameworks = null;
+                    if (!mappings.TryGetPortableFrameworks(targetFrameworkProfile, out clientFrameworks))
+                    {
+                        result = UnsupportedFramework;
+                    }
+                    else
+                    {
+                        var profileNumber = -1;
+                        if (mappings.TryGetPortableProfile(clientFrameworks, out profileNumber))
+                        {
+                            profile = FrameworkNameHelpers.GetPortableProfileNumberString(profileNumber);
+
+                        }
+                    }
+                }
+
+                if (!string.IsNullOrEmpty(targetPlatformVersion))
+                {
+                    if (targetPlatformVersion.IndexOf('.') < 0)
+                    {
+                        targetPlatformVersion += ".0";
+                    }
+
+                    if (!Version.TryParse(targetPlatformVersion, out platformVersion))
+                    {
+                        throw new ArgumentException(string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.InvalidFrameworkVersion,
+                            targetPlatformVersion));
+                    }
+                }
+
+                if (version.Major >= 5
+                    && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, framework))
+                {
+                    result = new NuGetFramework(framework, version, targetPlatformIdentifier ?? string.Empty, platformVersion);
```

Comment:
does the target platform identifier need normalized here? 
ToLowercase same as the others?

Reply:
Sure!

Reply:
wait, what do you mean by "the others"? I don't see this lowercasing happening anywhere.

Reply:
By others, I mean other parsing methods. However after a 2nd look, we don't do that at parse, we do it at `GetShortFolderName`, https://github.com/NuGet/NuGet.Client/blob/9d476f1b7453bbce324e2d7fe4a247b537c7121f/src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs#L286, so I think my recall was incorrect.

Reply:
Yeah I think we generally want to mostly keep the incoming formatting for frameworks, since they'll often have their "standard" capitalization, and I think it's nicer to output that.